src/server/routes/api/document/index.js

```diff
@@ -1259,11 +1266,25 @@ http.patch('/status/:id/:secret', function( req, res, next ){
   const handlePublishRequest = async doc => {
     const didPublish = await tryPublish( doc );
     if( didPublish ) {
+      updateRelatedPapers( doc );
       await tryTweetingDoc( doc );
       await sendFollowUpNotification( doc );
     }
   };
 
+  const updateRelatedPapers = doc => {
+    let docId = doc.id();
+    logger.info('Searching the related papers table for document ', docId);
+    try {
+      searchRelatedPapers( doc )
+        .then( papersData => createRelatedPapers({ papersData, docId }) )
+        .then( () => logger.info('Related papers table is updated for document', docId) );
+    } catch ( e ) {
```

Comment:
> Tag each related paper 'doc' with an interaction id or figure out some way to determine which papers to show for each Biofactoid interaction

The simplest approach is to store related papers for an interaction within the interaction.

Reply:
>Tag each related paper 'doc' with an interaction id or figure out some way to determine which papers to show for each Biofactoid interaction

Will this be done only when the related-papers is searched for a single interaction of a document?

>The simplest approach is to store related papers for an interaction within the interaction.

If it will be done in this way and anser to my question above is "yes" then would we handle it when we are integrating the trigger for searching relating papers for an interaction?

Reply:
>The simplest approach is to store related papers for an interaction within the interaction.

SInce we are already adding the relatedPapers to the database (https://github.com/PathwayCommons/factoid/pull/727/files#diff-9d36d49f610f5307b10c657fc6e1c3a8R84) would not we add a elID field along with the docID field? 

I do not know the use case where the related papers will be queried for an interaction. If it will be more convenient store them in interaction based on how they will be queried then we can do the same for the document as well. I mean we can store the related papers for a document in the document itself rather then keeping a relatedPapers table.

I guess it would be confusing to handle one from a seperate table but handling the other from the object.

Reply:
See above, I can't see a case where a `interactionId` parameter will be used. Assume the user submit is the only trigger.